ui/pages/confirmations/components/confirm/info/typed-sign/permit-simulation/permit-simulation.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { NameType } from '@metamask/name-controller';
+
+import { Numeric } from '../../../../../../../../shared/modules/Numeric';
+import Name from '../../../../../../../components/app/name/name';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowText,
+} from '../../../../../../../components/app/confirm/info/row';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../../../selectors';
+import { Box, Text } from '../../../../../../../components/component-library';
+import {
+  BackgroundColor,
+  BorderRadius,
+  Display,
+  TextAlign,
+} from '../../../../../../../helpers/constants/design-system';
+import { SignatureRequestType } from '../../../../../types/confirm';
+import useTokenExchangeRate from '../../../../../../../components/app/currency-input/hooks/useTokenExchangeRate';
+import { IndividualFiatDisplay } from '../../../../simulation-details/fiat-display';
+
+const PermitSimulation: React.FC = () => {
+  const t = useI18nContext();
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as SignatureRequestType;
+  const [faitValue, setFiatValue] = useState<number>();
+
+  const {
+    domain: { verifyingContract },
+    message: { value },
+  } = JSON.parse(currentConfirmation?.msgParams?.data as string);
+
+  const exchangeRate = useTokenExchangeRate(verifyingContract);
+
+  useEffect(() => {
+    if (exchangeRate && value) {
+      setFiatValue(exchangeRate.times(new Numeric(value, 10)).toNumber());
+    }
+  }, [exchangeRate, value]);
+
+  return (
+    <Box
+      backgroundColor={BackgroundColor.backgroundDefault}
+      borderRadius={BorderRadius.MD}
+      padding={2}
+      marginBottom={4}
+    >
+      <ConfirmInfoRow
+        label={t('simulationDetailsTitle')}
+        tooltip={t('simulationDetailsTitleTooltip')}
+      >
+        <ConfirmInfoRowText text={t('permitSimulationDetailInfo')} />
+      </ConfirmInfoRow>
+      <ConfirmInfoRow label={t('approve')}>
+        <Box>
+          <Box display={Display.Flex}>
+            <Box display={Display.Inline} marginRight={1}>
+              <Text
+                backgroundColor={BackgroundColor.backgroundAlternative}
+                borderRadius={BorderRadius.XL}
+                paddingInline={2}
+                textAlign={TextAlign.Center}
+              >
+                {value}
+              </Text>
+            </Box>
+            <Name value={verifyingContract} type={NameType.ETHEREUM_ADDRESS} />
+          </Box>
+          <Box>
+            {faitValue && <IndividualFiatDisplay fiatAmount={faitValue} />}
```

Comment:
I believe displaying the fiatValue would be subject to the user's settings. If so, I think we'll want to use getShouldShowFiat here

Reply:
Hey @SayaGT : can you plz suggest if user preference should be used here ?

Reply:
@digiwand : more details here https://consensys.slack.com/archives/C070G0RCX0T/p1717690653558139?thread_ts=1717689919.692409&cid=C070G0RCX0T

Reply:
Yes we should show the fiat value here regardless of users setting. We don't want to show any fiat value for test tokens though.

Reply:
I agree it's always fiat regardless of the fiat vs ETH setting. Something we might want to account for is that the existing simulation component takes this currency conversion setting in consideration:
<img width="339" alt="Screenshot 2024-06-06 at 14 14 03" src="https://github.com/MetaMask/metamask-extension/assets/34968356/3463e0fc-457e-41f5-8eb5-67af8624d990">

Reply:
thanks all! Per conversation, the following setting determines whether the simulation is shown or not, not the getShouldShowFiat with the getUseCurrencyRateCheck setting I was concerned about.

![CleanShot 2024-06-08 at 01 31 08@2x](https://github.com/MetaMask/metamask-extension/assets/20778143/3c450641-3277-4140-8f7d-bf04f24b223b)

Reply:
thanks all! Per conversation, the following setting determines whether the simulation is shown or not, not the getShouldShowFiat with the getUseCurrencyRateCheck setting I was concerned about.

![CleanShot 2024-06-08 at 01 31 08@2x](https://github.com/MetaMask/metamask-extension/assets/20778143/3c450641-3277-4140-8f7d-bf04f24b223b)

Reply:
to confirm, does the code hide the simulation when this is off?

Reply:
yes, we should hide simulations when this is off.

Reply:
I am creating a followup PR to hide simulation section behind feature flag for both Permit and SIWE signatures.